sat_vapor_pres/sat_vapor_pres_k.F90

```diff
@@ -1730,290 +3533,739 @@ end subroutine lookup_es2_k_0d
 !#######################################################################
 
  subroutine lookup_es3_des3_k_3d (temp, esat, desat, nbad)
- real, intent(in),  dimension(:,:,:)  :: temp
- real, intent(out), dimension(:,:,:)  :: esat, desat
+ class(*), intent(in),  dimension(:,:,:)  :: temp
+ class(*), intent(out), dimension(:,:,:)  :: esat, desat
  integer, intent(out)                 :: nbad
 
  real    :: tmp, del
  integer :: ind, i, j, k
+ logical :: valid_types = .false.
+
+   select type (temp)
+   type is (real(kind=r4_kind))
+     select type (esat)
+     type is (real(kind=r4_kind))
+       select type (desat)
+       type is (real(kind=r4_kind))
+         valid_types = .true.
+       end select
+     end select
+   type is (real(kind=r8_kind))
+     select type (esat)
+     type is (real(kind=r8_kind))
+       select type (desat)
+       type is (real(kind=r8_kind))
+         valid_types = .true.
+       end select
+     end select
+   end select
+   if ( .not. valid_types ) then
+     call error_mesg ('sat_vapor_pres_k_mod::lookup_es3_des3_k_3d', 'unsupported kind', FATAL)
```

Comment:
This error message should be more descriptive.  *"temp, desat, and esat types do not match .or. desat and/or esat is not one of the supported types of real(kind=4) or real(kind=8)."*